### ISSUE TITLE
Phase 3c: ONNX runtime for customer-trained TLS classifiers

### ISSUE THREAD

**Issue by fabriziosalmi** [fabriziosalmi/zion#29]
Adds support for loading customer-trained TLS-fingerprint classifier models at runtime, via the [`tract`](https://github.com/sonos/tract) ONNX inference engine. Builds on Phase 3b (#28) by replacing — or running alongside — the binary-baked logistic regression with a model the customer has trained on their own traffic.

This is the "bring your own model" tier. Phase 3b ships the OSS-default model trained on public datasets; Phase 3c lets a customer export a model from their own ML pipeline and point Zion at it.

## Why a customer would want this

The OSS-default model in Phase 3b is trained on CIC-IDS / CTU-13 — public, US-skewed, 2017–2018 vintage. For a deployment with:

* Non-default geographic distribution (the model's class boundaries don't reflect the local long tail)
* Domain-specific traffic patterns (B2B API, IoT telemetry, mobile-only app — ClientHello distributions differ from generic web)
* Internal threat intel (the customer has labeled-malicious samples Zion's authors don't have)
* Compliance requirement to use a model the customer can audit end-to-end

…the right answer is the customer's model, not Zion's.

## Architecture

Configured under the same `[tls.fingerprint]` section as Phase 3b:

```toml
[tls.fingerprint]
mode  = "score"
model = "onnx:/etc/zion/customer-bot.onnx"   # vs "builtin-v1" in Phase 3b

# Optional model metadata for `/metrics` and audit:
model_version = "customer-2026Q1"
model_threshold = 0.5
```

At startup:
1. Parse `model = "onnx:..."` → load ONNX file via `tract`.
2. Validate input shape: must be a `[1, N]` `f32` tensor where `N` matches Zion's feature extractor output (~50–80 dimensions). Refuse to start with a structured error if the shape is wrong — saves a runtime crash.
3. Validate output shape: must be `[1, K]` for `K ∈ {1, 2, 3, 4}` (binary, or up to 4-way class).
4. Bake the model into an `Arc<TypedRunnableModel>` and store it in `state.config` alongside the rest of the snapshot.

Hot path:
1. Same feature extraction as Phase 3b.
2. Replace `logistic_regression(features)` with `tract_model.run(tensor!(f32; ...))`.
3. Costs: tract Inference Engine ~3–10 µs depending on model size (vs ~100 ns for baked logistic). Still ≤ 5 % of TLS handshake — acceptable.

## Hot-reload behaviour

The ONNX file path is part of `[tls.fingerprint]`. The Phase 1 file watcher already triggers a reload on `zion.toml` changes. Phase 3c adds:

* If `model = "onnx:PATH"` and `PATH` itself changes on disk → reload (new file watcher subscription).
* On-error: log REJECT and keep the previously-loaded model. Same contract as the rest of Phase 1 hot-reload.
* On model swap: in-flight request finishes with the old model snapshot; next request gets the new one. Same `Arc<...>` snapshot model that Phase 1 introduced.

## Roadmap (incremental commits)

1. New cargo feature `tls-fingerprint-onnx` (gated; default off). Adds `tract` dependency only when the feature is on. Customers who don't use ONNX get zero binary-size impact.
2. Model loader: `src/tls_fp_onnx.rs` reads the file, validates shapes, returns `Arc<TypedRunnableModel>` (tract type).
3. Wire-up in `[tls.fingerprint]` config parser: `model = "onnx:PATH"` is accepted; `builtin-v1` continues to work; ambiguity → reject at boot.
4. Inference path: switch the dispatcher in Phase 3b to use whichever model is loaded; same header schema unchanged.
5. File watcher subscription for the ONNX file (separate from `zion.toml` itself).
6. Docs (`docs/security/tls-scoring-onnx.md`) covering: training pipeline, ONNX export from sklearn / PyTorch / Keras, validation procedure (Zion-supplied `zion validate-onnx /path/to/model.onnx` CLI command).
7. Integration test: a tiny ONNX model (5-feature logistic in ONNX format) committed to `tests/fixtures/`, soak-tested against canned ClientHello samples.

Total estimated diff: ~500 LoC Rust + ~300 LoC test fixtures + ONNX validation tool, **+1 dependency** (`tract` ~25 MB compiled into the binary; gated by feature so default builds are unaffected).

## Acceptance criteria

* [ ] `cargo build --release` (no `tls-fingerprint-onnx` feature): binary size unchanged, no `tract` symbols linked.
* [ ] `cargo build --release --features tls-fingerprint-onnx`: builds clean, binary grows by ~25 MB.
* [ ] `model = "onnx:/path/that/does/not/exist"` at boot: structured error, daemon refuses to start (vs silently downgrading to builtin-v1, which would be a security risk).
* [ ] `model = "onnx:/path/with/wrong/input/shape"` at boot: structured error including expected shape vs actual.
* [ ] At runtime, hot-reload of an ONNX file produces a "model swapped" log line and bumps `zion_tls_model_version` Prometheus gauge to the new `model_version`.
* [ ] An ONNX file with a malformed binary content: REJECT, previous model survives, log line `model_load: REJECTED (...), keeping previous`.
* [ ] CLI tool `zion validate-onnx <path>` prints expected vs actual input shape and a verdict before the operator commits to deploying the file.
* [ ] Inference cost measured: < 50 µs P99 for models up to 100K parameters (the realistic ceiling for "superlight" classifiers customers should ship).
* [ ] Documentation explicit that ONNX models are NOT a free lunch — they bring with them all the obligations of model maintenance (drift, retrain cadence, explainability) onto the customer's plate.

## Out of scope

* GPU-accelerated inference (CUDA / Metal).
* Online learning, federated learning, A/B testing of multiple models.
* Model authoring tooling — Zion validates and runs; customer trains.
* Hosting / signing / verification of customer ONNX models. The customer is responsible for the integrity of the file they put in `/etc/zion/`.
* Quantised models — `tract` supports them but exposing the knobs adds API surface; revisit if a real customer asks.

## Risks

1. **`tract` is a sizable dependency.** ~25 MB compiled. Mitigation: behind a cargo feature, default off. Customers who don't need ONNX get the lean binary.
2. **ONNX standard has corners.** Not every ONNX op is supported by `tract`. Mitigation: documented compatibility matrix at `docs/security/tls-scoring-onnx.md`; the `zion validate-onnx` CLI surfaces unsupported ops at validate time, not at inference time.
3. **Adversarial models from a careless training pipeline.** A customer who trains on biased labels gets a biased model. Mitigation: docs are explicit that the score is the customer's responsibility; Zion ships the framework, not the verdict.
4. **Inference latency variance.** Some ONNX models have data-dependent execution paths (decision trees with early exit, RNN). Mitigation: validate-onnx runs N=1000 samples, reports P50/P99 latency, refuses to install if P99 > 50 µs.

## References

* `tract` (Sonos's Rust ONNX engine): https://github.com/sonos/tract
* ONNX spec: https://onnx.ai/
* Phase 3a (allowlist): #27
* Phase 3b (built-in scoring): #28
* Architectural decision why "score, not gate" stays unchanged: see #28 § "Decision rule, downstream"
